processing/src/main/java/org/apache/druid/segment/filter/BoundFilter.java

```diff
@@ -106,22 +106,27 @@ public BitmapColumnIndex getBitmapColumnIndex(ColumnIndexSelector selector)
       }
       final NumericRangeIndex rangeIndex = indexSupplier.as(NumericRangeIndex.class);
       if (rangeIndex != null) {
-        final Number lower = boundDimFilter.hasLowerBound() ? Double.parseDouble(boundDimFilter.getLower()) : null;
-        final Number upper = boundDimFilter.hasUpperBound() ? Double.parseDouble(boundDimFilter.getUpper()) : null;
-        final BitmapColumnIndex rangeBitmaps = rangeIndex.forRange(
-            lower,
-            boundDimFilter.isLowerStrict(),
-            upper,
-            boundDimFilter.isUpperStrict()
-        );
-        if (rangeBitmaps != null) {
-          // preserve sad backwards compatible behavior where bound filter matches 'null' if the lower bound is not set
-          if (boundDimFilter.hasLowerBound() && !NullHandling.isNullOrEquivalent(boundDimFilter.getLower())) {
-            return rangeBitmaps;
-          } else {
-            return wrapRangeIndexWithNullValueIndex(indexSupplier, rangeBitmaps);
+        try {
+          final Number lower = boundDimFilter.hasLowerBound() ? Double.parseDouble(boundDimFilter.getLower()) : null;
+          final Number upper = boundDimFilter.hasUpperBound() ? Double.parseDouble(boundDimFilter.getUpper()) : null;
+          final BitmapColumnIndex rangeBitmaps = rangeIndex.forRange(
+              lower,
+              boundDimFilter.isLowerStrict(),
+              upper,
+              boundDimFilter.isUpperStrict()
+          );
+          if (rangeBitmaps != null) {
+            // preserve sad backwards compatible behavior where bound filter matches 'null' if the lower bound is not set
+            if (boundDimFilter.hasLowerBound() && !NullHandling.isNullOrEquivalent(boundDimFilter.getLower())) {
+              return rangeBitmaps;
+            } else {
+              return wrapRangeIndexWithNullValueIndex(indexSupplier, rangeBitmaps);
+            }
           }
         }
+        catch (NumberFormatException ignored) {
+          // bounds are not numeric?
```

Comment:
The `catch` here doesn't really sit right. Where does the `NumberFormatException` get thrown? Is it possible to deal with that case some other way?

My concern here is that there's a lot of code and method calls wrapped into that `try` block, and we have no way of telling where the `NumberFormatException` came from. Maybe scoping the `try` down to a smaller bit of code would help with that.

Reply:
They come from the `Double.parseDouble` in
```
final Number lower = boundDimFilter.hasLowerBound() ? Double.parseDouble(boundDimFilter.getLower()) : null;
final Number upper = boundDimFilter.hasUpperBound() ? Double.parseDouble(boundDimFilter.getUpper()) : null;
```
will see if i can shuffle some stuff around